src/loaders/earthEngineLoader.js

```diff
@@ -1,196 +1,29 @@
-                },
-                {
-                    color: '#399b38',
-                    name: i18n.t('Deciduous Needleleaf forest'),
-                },
-                {
-                    color: '#38eb38',
-                    name: i18n.t('Deciduous Broadleaf forest'),
-                },
-                {
-                    color: '#39723b',
-                    name: i18n.t('Mixed forest'),
-                },
-                {
-                    color: '#6a2424',
-                    name: i18n.t('Closed shrublands'),
-                },
-                {
-                    color: '#c3a55f',
-                    name: i18n.t('Open shrublands'),
-                },
-                {
-                    color: '#b76124',
-                    name: i18n.t('Woody savannas'),
-                },
-                {
-                    color: '#d99125',
-                    name: i18n.t('Savannas'),
-                },
-                {
-                    color: '#92af1f',
-                    name: i18n.t('Grasslands'),
-                },
-                {
-                    color: '#10104c',
-                    name: i18n.t('Permanent wetlands'),
-                },
-                {
-                    color: '#cdb400',
-                    name: i18n.t('Croplands'),
-                },
-                {
-                    color: '#cc0202',
-                    name: i18n.t('Urban and built-up'),
-                },
-                {
-                    color: '#332808',
-                    name: i18n.t('Cropland/Natural vegetation mosaic'),
-                },
-                {
-                    color: '#d7cdcc',
-                    name: i18n.t('Snow and ice'),
-                },
-                {
-                    color: '#f7e174',
-                    name: i18n.t('Barren or sparsely vegetated'),
-                },
-                {
-                    color: '#743411',
-                    name: i18n.t('Unclassified'),
-                },
-            ],
-        },
-        popup: '{name}: {value}',
-    },
-});
+import { getInstance as getD2 } from 'd2';
+import { getEarthEngineLayer } from '../constants/earthEngine';
+import { getPeriodFromFilter } from '../util/earthEngine';
+import { getOrgUnitsFromRows } from '../util/analytics';
+import { getDisplayProperty } from '../util/helpers';
+import { toGeoJson } from '../util/map';
 
 // Returns a promise
 const earthEngineLoader = async config => {
-    const datasets = getDatasets();
+    const { rows, aggregationType } = config;
+    const orgUnits = getOrgUnitsFromRows(rows);
     let layerConfig = {};
     let dataset;
+    let features;
+
+    if (orgUnits && orgUnits.length) {
+        const d2 = await getD2();
+        const displayProperty = getDisplayProperty(d2).toUpperCase();
+        const orgUnitParams = orgUnits.map(item => item.id);
+
+        features = await d2.geoFeatures
```

Comment:
We should probably move this to `@dhis2/analytics` similar to the `analytics` part.

Reply:
Agreed :-)